src/linker/configuration.py

```diff
@@ -17,30 +17,32 @@ class Config:
 
     def __init__(
         self,
-        pipeline_specification: Optional[Path],
-        input_data: Optional[Path],
-        computing_environment: Optional[str],
+        pipeline_specification: Path,
+        input_data: Path,
+        computing_environment: Optional[Path],
     ):
-        self.pipeline_path = pipeline_specification
-        self.pipeline = load_yaml(self.pipeline_path) if pipeline_specification else {}
-        self.input_data = self._load_input_data_paths(input_data) if input_data else []
-        self.computing_environment_path = (
-            Path(computing_environment) if computing_environment else None
+        # Handle pipeline specification
+        self.pipeline_specification_path = pipeline_specification
+        self.pipeline = load_yaml(self.pipeline_specification_path)
+        # Handle input data specification
+        self.input_data_specification_path = input_data
+        self.input_data = self._load_input_data_paths(self.input_data_specification_path)
+        # Handle environment specification
+        self.computing_environment_specification_path = computing_environment
+        self.environment = self._load_computing_environment(
+            self.computing_environment_specification_path
         )
-        self.environment = self._load_computing_environment(self.computing_environment_path)
-
         self.computing_environment = self.environment["computing_environment"]
-        self.container_engine = self.environment.get("container_engine", "undefined")
+        self.container_engine = self.environment["container_engine"]
+        self.implementation_resources = self.environment.get("implementation_resources", {})
+        self.slurm = self.environment.get("slurm", {})
+        self.slurm_resources = self._get_slurm_resources()
         self.spark = self._get_spark_requests(self.environment)
+        self.spark_resources = {key: getattr(self, key) for key in ["slurm", "spark"]}
```

Comment:
As I mentioned in a comment elsewhere, I don't love the creation and then usage of this somewhat complicated and bespoke dictionary. It'd be better imo to expose properties or methods to get the specific parts of the config at the point of usage.